cli/src/cmd/forge/script/mod.rs

```diff
@@ -0,0 +1,343 @@
+mod build;
+use build::BuildOutput;
+
+mod runner;
+use runner::Runner;
+
+mod broadcast;
+use broadcast::into_legacy;
+use ui::{TUIExitReason, Tui, Ui};
+
+mod cmd;
+
+mod executor;
+
+use crate::{cmd::forge::build::BuildArgs, opts::MultiWallet};
+use clap::{Parser, ValueHint};
+use ethers::{
+    abi::{Abi, Function, RawLog},
+    prelude::{artifacts::ContractBytecodeSome, ArtifactId, Bytes, Project},
+    types::{transaction::eip2718::TypedTransaction, Address, TransactionRequest, U256},
+};
+use forge::{
+    debug::DebugArena,
+    decode::decode_console_logs,
+    executor::opts::EvmOpts,
+    trace::{
+        identifier::{EtherscanIdentifier, LocalTraceIdentifier},
+        CallTraceArena, CallTraceDecoder, CallTraceDecoderBuilder, TraceKind,
+    },
+};
+
+use foundry_common::evm::EvmArgs;
+use foundry_config::Config;
+use foundry_utils::{encode_args, format_token, IntoFunction};
+
+use std::{
+    collections::{BTreeMap, VecDeque},
+    path::PathBuf,
+    time::Duration,
+};
+
+use yansi::Paint;
+
+// Loads project's figment and merges the build cli arguments into it
+foundry_config::impl_figment_convert!(ScriptArgs, opts, evm_opts);
+
+#[derive(Debug, Clone, Parser)]
+pub struct ScriptArgs {
+    /// The path of the contract to run.
+    ///
+    /// If multiple contracts exist in the same file you must specify the target contract with
+    /// --target-contract.
+    #[clap(value_hint = ValueHint::FilePath)]
+    pub path: PathBuf,
+
+    /// Arguments to pass to the script function.
+    pub args: Vec<String>,
+
+    /// The name of the contract you want to run.
+    #[clap(long, alias = "tc")]
+    pub target_contract: Option<String>,
+
+    /// The signature of the function you want to call in the contract, or raw calldata.
+    #[clap(long, short, default_value = "run()")]
+    pub sig: String,
+
+    #[clap(
+        long,
+        help = "Use legacy transactions instead of EIP1559 ones. this is auto-enabled for common networks without EIP1559."
+    )]
+    pub legacy: bool,
+
+    #[clap(long, help = "Broadcasts the transactions.")]
+    pub broadcast: bool,
+
+    #[clap(flatten, next_help_heading = "BUILD OPTIONS")]
+    pub opts: BuildArgs,
+
+    #[clap(flatten)]
+    pub wallets: MultiWallet,
+
+    #[clap(flatten, next_help_heading = "EVM OPTIONS")]
+    pub evm_opts: EvmArgs,
+
+    /// Resumes submitting transactions that failed or timed-out previously.
+    ///
+    /// It DOES NOT simulate the script again and it expects nonces to have remained the same.
+    ///
+    /// Example: If transaction N has a nonce of 22, then the account should have a nonce of 22,
+    /// otherwise it fails.
+    #[clap(long)]
+    pub resume: bool,
+
+    /// Resumes submitting transactions that failed or timed-out previously. SHOULD NOT be used
+    /// when deploying CREATE contracts.
```

Comment:
as opposed to CREATE2

CREATE depends on nonces and since `--force-resume` does not care for the initial nonce, the deployed contracts would have different addresses than expected.

It's possible that `--force-resume` is just too niche and may cause confusion, so i'm fine with removing it

Reply:
yep - let's keep only 1 resume option